compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -228,6 +233,14 @@ class TypeOps:
           val ssym = sym.symbolName
           s.TypeRef(spre, ssym, Seq.empty)
 
+        case TypeRef(pre, sym: Name) =>
+          // val spre = if tpe.hasTrivialPrefix then s.Type.Empty else loop(pre)
+          // Craft semanticdb symbol
+          // prefix symbol (e.g. "scala/") + name (e.g. "Nothing") + Global symbol suffix ("#")
+          // see: https://scalameta.org/docs/semanticdb/specification.html#symbol
+          val ssym = s"${pre.typeSymbol.symbolName}${sym.mangledString}#"
```

Comment:
here, could you try to call perhaps `symbol` or `denot` on the type ref to compute a symbol?

Reply:
As far as I know, we can't. Because here, `TypeRef` doesn't have `Symbol`: `case TypeRef(pre, sym: Name) =>`.
Is there any way to compute `Symbol` from `Name` from here?

Reply:
those are both methods on `NamedType`, which `TypeRef` is a subclass of - they compute a symbol by looking up in the context of the prefix - alternatively this could be another situation to make a `FakeSymbol`

Reply:
While the designator is `Name`, realized we can access the symbol for `TypeRef` itself via https://github.com/lampepfl/dotty/blob/4fa62e22d08ff79468a5802209625f1b45623c8a/compiler/src/dotty/tools/dotc/core/Types.scala#L2157-L2160 :)
https://github.com/lampepfl/dotty/pull/14608/commits/4fa62e22d08ff79468a5802209625f1b45623c8a